### PR TITLE
enable new property ov::hint::use_cpu_pinning

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *enable use_cpu_pinning for inference. Developers can choose whether to bind OV threads to CPU processors during inference*
 - *Will integration this function into CPU inference flow in another PR*

### Tickets:
 - *106349*

### CODE REVIEW

File: src/inference/include/openvino/runtime/properties.hpp [hunk 1]
```diff
@@ -351,6 +351,22 @@ inline std::istream& operator>>(std::istream& is, PerformanceMode& performance_m
...
+ * value for this property, OpenVINO may choose any desired value based on internal logic.
+ *
+ * The following code is example to use this property.
+ *
+ * @code
+ * ie.set_property(ov::hint::use_cpu_pinning(true));
+ * ie.set_property(ov::hint::use_cpu_pinning(false));
+ * @endcode
+ */
+static constexpr Property<bool> use_cpu_pinning{"USE_CPU_PINNING"};
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/properties.hpp:368]: is it supposed that we need to deprecate AFFINITY?

Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp:368]: Yes, we need to document AFFINITY deprecation in 2023.0 and deprecate AFFINITY in 2024.0.

Comment (ilya-lavrenov) [reply src/inference/include/openvino/runtime/properties.hpp:368]: Why in 2024?
We can deprecate even now.

Is it expected to fully replace affinity by use_cpu_pinning ? Affinity has more values like NUMA, CORES than just true / false in use_cpu_pinning

Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp:368]: Could we deprecate property without information in advance? 

From technique perspective, yes, we will replace affinity by use_cpu_pinning now.

**Review by sunxiaoxia2022 — APPROVED** [review]

File: src/plugins/intel_cpu/src/config.h [hunk 2]
```diff
@@ -49,6 +49,8 @@ struct Config {
     size_t rtCacheCapacity = 5000ul;
     InferenceEngine::IStreamsExecutor::Config streamExecutorConfig;
     InferenceEngine::PerfHintsConfig  perfHintsConfig;
+    bool use_pin_value = true;
+    bool use_pin_changed = false;
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/config.h:53]: Please look on code style comment from other PRs.

**Comment by dmitry-gorokhov** [comment]
@wangleis General question: what will be the mapping on legacy affinity property? `use_cpu_pinning=true` equals `CORE` affinity and `use_cpu_pinning=false` equals `NUMA/HybridAware` affinity? The case there we disable pinning at all will not be supported?

**Comment by wangleis** [comment]
@dmitry-gorokhov The mapping between legacy affinity property and pinning property will be included in separated PRs which integration new properties into new inference flow. 

We will propose `use_cpu_pinning` is higher priority than `affinity`. `use_cpu_pinning=true` equals `CORE/HybridAware` affinity and `use_cpu_pinning=false` equals NUMA affinity. If user set inconsistent value to two properties, we will use `use_cpu_pinning` value and update `affinity` value.

**Comment by wangleis** [comment]
@dmitry-gorokhov @ilya-lavrenov Updated for conflict due to ov::hint::use_hyper_threading and ov::hint::scheduling_core_type are merged.

Please review and approve this PR. Thanks.

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
We still agreed to deprecate affinity property. Lets proceed with that.

**Comment by wangleis** [comment]
> We still agreed to deprecate affinity property. Lets proceed with that.

Yes, will handle affinity property in the following PR which integrate CPU pinning property into flow.

File: src/inference/include/openvino/runtime/properties.hpp [hunk 3]
```diff
@@ -409,10 +409,26 @@ inline std::istream& operator>>(std::istream& is, SchedulingCoreType& core_type)
...
  * @ingroup ov_runtime_cpp_prop_api
  *
- * Developer can use this property to use or not use hyper threading during inference. If user does not explicitly set
+ * Developer can use this property to use or not use CPU threads pinning during inference. If user does not explicitly
+ * set value for this property, OpenVINO may choose any desired value based on internal logic.
+ *
+ * The following code is example to use this property.
+ *
+ * @code
+ * ie.set_property(ov::hint::use_cpu_pinning(true));
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/properties.hpp:421]: BTW, all other options are called `enable_` something:
```c++
enable_startup_fallback
enable_execution_fallback
enable_mmap
enable_hyper_threading
enable_profiling
enable_loop_unrolling
```
I proper to have an alignment here

Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp:421]: > BTW, all other options are called `enable_` something:
> 
> ```c++
> enable_startup_fallback
> enable_execution_fallback
> enable_mmap
> enable_hyper_threading
> enable_profiling
> enable_loop_unrolling
> ```
> 
> I propose having an alignment here

Ok. Will create follow up PR to update use_cpu_pinning and use_hyper_threading

Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp:421]: https://github.com/openvinotoolkit/openvino/pull/16723

**Pull request merged**